### PR TITLE
Add template for multilingual hybrid doc search

### PR DESCRIPTION
**Pull request by sjrl**
@robpasternak has been investigating more retrieval and re-ranking models and he has recently tried out some multilingual models that have worked really well on the German DPR evaluation set. In this case, the multilingual retriever model has shown to outperform all german specific retrieval models we have tried to date, and the reranker model is competitive with the best german reranker we have to date. 

Rob has collected results for retriever comparisons on German DPR which can be found [here](https://github.com/deepset-ai/zeit-online-retriever/blob/main/results/gdpr_results.csv). Best performing models (from the `multilingual-e5-` family) are shown at the bottom of the table. And here are some of the results from that link.

### German DPR Results
| Model Name | Recall @ 1 | MRR @ 10|
|--------|--------|--------|
| paraphrase-multilingual-mpnet-base-v2 | 0.487 | 0.625 |
| bi-encoder_msmarco_bert-base_german | 0.586 | 0.719 |
| multilingual-e5-base | 0.623 | 0.763 |
| multilingual-e5-large | 0.685 | 0.802 |

Since these models are multilingual we thought it would be helpful to create a multilingual Doc Search template that we could reference as solid starting point for supporting languages other than English and German when doing doc retrieval. 

## Models
* The chosen embedding retrieve model, [multilingual-e5-base](https://huggingface.co/intfloat/multilingual-e5-base) supports upwards of 90 languages. 
* The chosen reranker model [cross-encoder/mmarco-mMiniLMv2-L12-H384-v1](https://huggingface.co/cross-encoder/mmarco-mMiniLMv2-L12-H384-v1) supports 14 languages. 

## Additional Notes
* The total overlap in languages between the retriever and reranker is 13 and they are: ar, de, en, es, fr, hi, it, ja, nl, pt, ru, vi, zh
* Of the 13 languages the sentence splitter in the Preprocessor only supports the languages de, en, fr, it, nl, pt, ru, so we could opt to turn off sentence splitting by default in this template to avoid needing to specify a language in the Preprocessor.

### CODE REVIEW

**Review by agnieszka-m — APPROVED** [review]

**Comment by sjrl** [comment]
We would need to update the cross encoder to https://huggingface.co/jeffwan/mmarco-mMiniLMv2-L12-H384-v1

**Comment by sjrl** [comment]
I think this template would be very helpful for a few reasons:
* It enables cross-lingual search. So user types in German and can get back relevant English docs. This is a use case that @JasperLS has come across.
* This is also a great starting point for clients who have documents in a language other than English. So this should work out of the box fairly well for German, French, Spanish, etc.

**Review by erendabanlioglu — APPROVED** [review]
To make this template available in dc we should also update UI repo. Please align with @mathislucka and agree on it. FYI @oryx1729

File: pipelines/HybridDocSearch_multilingual.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# If you need help with the YAML format, have a look at https://docs.cloud.deepset.ai/docs/create-a-pipeline#create-a-pipeline-using-yaml.
+# This is a friendly editor that helps you create your pipelines with autosuggestions. To use them, press Control + Space on your keyboard.
+# Whenever you need to specify a model, this editor helps you out as well. Just type your Hugging Face organization and a forward slash (/) to see available models.
+
+# This is a document search pipeline that works on multiple languages that combines vector-based and keyword-based searches.
+# Such combination usually yields the best results without any training.
+# The supported languages are: ar, de, en, es, fr, hi, it, ja, nl, pt, ru, vi, zh
+version: '1.21.2'
+name: 'HybridDocumentSearch_multilingual'
```
Comment (ArzelaAscoIi) [inline pipelines/HybridDocSearch_multilingual.yaml:9]: @yaquelinehoyos Did we already shipped the new pipeline creation flow ?

**Comment by erendabanlioglu** [comment]
There is [PR](https://github.com/deepset-ai/haystack-hub-ui/pull/2426) to add this template @sjrl but @agnieszka-m had question if you can take a look.

**Pull request merged**